packages/babel-generator/src/generators/statements.js

```diff
@@ -123,10 +123,14 @@ function buildLabelStatement(prefix, key = "label") {
     const label = node[key];
     if (label) {
       this.space();
-
-      const terminatorState = this.startTerminatorless();
-      this.print(label, node);
-      this.endTerminatorless(terminatorState);
+      if (prefix == "break" || prefix == "continue") {
+        // see https://git.io/vQ7uG
+        this.print(label, node);
```

Comment:
This avoids one issue, but won't 100% solve the issue. Consider if the label has a `leadingComment` of a `LineComment`. What is the right output?

```
continue foo;
```
becomes
```
continue // a comment
  foo;
```

instead of
```
continue (// a comment
  foo);
```
but it is still not right, since the spec doesn't allow newlines between `continue` and the label, so that is parsed like

```
continue;
// a comment
foo;
```

We'd almost have to move the comment output to _before_ `continue`, which will be a pain.